src/components/EventDashboard/ResourceTabPanel.tsx

```diff
@@ -0,0 +1,541 @@
+  numSelected: number;
+  onSelectAllClick: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  rowCount: number;
+  handleIncludeClick: () => void;
+  handleExcludeClick: () => void;
+}
+
+const EnhancedTableToolbar = (props: EnhancedTableToolbarProps) => {
+  const {
+    numSelected,
+    onSelectAllClick,
+    rowCount,
+    handleIncludeClick,
+    handleExcludeClick,
+  } = props;
+  const classes = useStyles();
+
+  return (
+    <Toolbar className={classes.toolbar} disableGutters>
+      <Checkbox
+        indeterminate={numSelected > 0 && numSelected < rowCount}
+        checked={rowCount > 0 && numSelected === rowCount}
+        onChange={onSelectAllClick}
+      />
+      <Typography variant="body1" color="secondary">
+        Select all
+      </Typography>
+      {numSelected > 0 && (
+        <Box
+          display="flex"
+          justifyContent="flex-end"
+          alignItems="center"
+          flexGrow={1}
+        >
+          <Typography
+            variant="body2"
+            color="textSecondary"
+            component="div"
+            style={{ marginRight: '16px' }}
+          >
+            {`${numSelected} selected`}
+          </Typography>
+          <Button color="secondary" onClick={handleIncludeClick}>
+            <Add className={classes.buttonIcon} />
+            Include
+          </Button>
+          <Button color="secondary" onClick={handleExcludeClick}>
+            <Remove className={classes.buttonIcon} />
+            Exclude
+          </Button>
+        </Box>
+      )}
+    </Toolbar>
+  );
+};
+
+const ResourceTabPanel = ({
+  eventId,
+  type,
+  hospitals = [],
+  ambulances = [],
+}: ResourceTabPanelProps) => {
+  const classes = useStyles();
+  const [order, setOrder] = React.useState<Order>('asc');
+  const [orderBy, setOrderBy] = React.useState<string>(
+    type === TabOptions.Hospital ? 'name' : 'vehicleNumber'
+  );
+  const [selected, setSelected] = React.useState<ResourceData[]>([]);
+  const [
+    singleSelected,
+    setSingleSelected,
+  ] = React.useState<ResourceData | null>(null);
+  const [open, setOpen] = React.useState<boolean>(false);
+  const [includeOrExclude, setIncludeOrExclude] = React.useState<
+    'include' | 'exclude' | null
+  >(null);
+
+  // Have to refetch GET_EVENT_BY_ID because only the list
+  // of all events in the cache is automatically updated
+  const mutationOptions = {
+    refetchQueries: [
+      {
+        query: GET_EVENT_BY_ID,
+        variables: { eventId },
+      },
+    ],
+  };
+
+  const [addHospitalsToEvent] = useMutation(
+    ADD_HOSPITALS_TO_EVENT,
+    mutationOptions
+  );
+  const [addAmbulancesToEvent] = useMutation(
+    ADD_AMBULANCES_TO_EVENT,
+    mutationOptions
+  );
+  const [deleteHospitalsFromEvent] = useMutation(
+    DELETE_HOSPITALS_FROM_EVENT,
+    mutationOptions
+  );
+  const [deleteAmbulancesFromEvent] = useMutation(
+    DELETE_AMBULANCES_FROM_EVENT,
+    mutationOptions
+  );
```

Comment:
I'm going to try to explain why I'm refetching the `GET_EVENT_BY_ID` query after each of these mutations.

This is the part of the cache that is relevant here:
![image](https://user-images.githubusercontent.com/22457859/89209350-1accf980-d58c-11ea-8529-daeb078b4c97.png)

These mutations will automatically update `events` in the cache **IF** the event with the correct eventId (i.e. `id = 3` in this case) is existing in the `events` field of the cache. However, I'm using `GET_EVENT_BY_ID` to get event specific data, which is added to the cache in the field `event({"id":"3"})`.

IDEALLY, we would use [readQuery](https://www.apollographql.com/docs/react/caching/cache-interaction/#readquery) so that the `GET_EVENT_BY_ID` queries the cache and not the server (and thus using the automatically updated data in the cache (I hope)). UNFORTUNATELY, `readQuery` throws an error if the cache doesn't contain all of the data necessary to fulfill a specified query. It never attempts to fetch data from a remote server. As such, it's not really usable for our use case here. I've spent a long time investigating this issue, and if anyone has a better solution, I'd be happy to try it out, but this is what I ended up implementing.

Reply:
SOLVED BY RETURNING ID in `GET_EVENT_BY_ID` although it still is kinda odd to me that the `GET_EVENT_BY_ID` result is cached separately and not inside the events field.